kubectl-pg/cmd/delete.go

```diff
@@ -0,0 +1,114 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1"
+	PostgresqlLister "github.com/zalando/postgres-operator/pkg/generated/clientset/versioned/typed/acid.zalan.do/v1"
+	"io/ioutil"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/scheme"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+// deleteCmd represents kubectl pg delete.
+var deleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Deletes postgresql object by object-name/manifest file",
+	Long: `Deletes the postgres objects specific to a manifest file or object-name.
+Deleting the manifest is sufficient to delete the cluster.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		namespace, _ := cmd.Flags().GetString("namespace")
+		file, _ := cmd.Flags().GetString("file")
+		if file != "" {
+			deleteByFile(file)
+		} else if namespace != "" {
+			if len(args) != 0 {
+				clusterName := args[0]
+				deleteByName(clusterName, namespace)
+			} else {
+				fmt.Println("cluster name can't be empty")
+			}
+		} else {
+			fmt.Println("use the flag either -n or -f to delete a resource.")
+		}
+	},
+	Example: "kubectl pg delete -f [FILE-NAME]\nkubectl pg delete [CLUSTER-NAME] -n [NAMESPACE]",
+}
+
+func init() {
+	namespace := getCurrentNamespace()
+	deleteCmd.Flags().StringP("namespace", "n", namespace, "Delete postgresql resource by it's name.")
+	deleteCmd.Flags().StringP("file", "f", "", "using file.")
+	rootCmd.AddCommand(deleteCmd)
+}
+
+
+func deleteByFile(file string) {
+	config := getConfig()
+	postgresConfig, err := PostgresqlLister.NewForConfig(config)
+	ymlFile, err := ioutil.ReadFile(file)
+	if err != nil {
+		panic(err)
+	}
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	obj, _, err := decode([]byte(ymlFile), nil, &v1.Postgresql{})
+	if err != nil {
+		panic(err)
+	}
+	postgresSql := obj.(*v1.Postgresql)
+	confirmDeletion(postgresConfig, postgresSql.Name, postgresSql.Namespace)
+}
+
+
+func deleteByName(clusterName string, namespace string) {
+	config := getConfig()
+	postgresConfig, err := PostgresqlLister.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+	confirmDeletion(postgresConfig, clusterName, namespace)
+}
+
+//Confirm delete & delete postgresql cluster.
+func confirmDeletion(postgresConfig *PostgresqlLister.AcidV1Client, clusterName string, namespace string) {
+	_, err := postgresConfig.Postgresqls(namespace).Get(clusterName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("-> postgresql %s not found with the provided namespace %s : %s \n", clusterName, namespace, err)
+		return
+	}
+enterClusterName:
```

Comment:
this confirmation loop is used at least twice (here and in confirm scaling) , move it to a separate function and re-write using the `for`-loop and hint a user on how to exit this loop without changing  the cluster

Reply:
I have moved this confirmation part to common place and changed goto to for loop. Now delete and scale will call confirmAction function from util.